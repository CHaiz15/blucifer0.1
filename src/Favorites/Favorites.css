.no-favorites {
  margin: 1% 0 0 3%;
  text-shadow: 0 0 1px #000;
  font-size: 1.25em;
}
