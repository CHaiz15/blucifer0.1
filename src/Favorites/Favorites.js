import React from 'react';
import PropTypes from 'prop-types';
import './Favorites.css'
import ListingCard from '../ListingCard/ListingCard.js'

const Favorites = ({ addSelectedListing, selectedAreaId, listings, favoriteListings }) => {
    const filteredListings = listings.filter(listing => favoriteListings.includes(listing.listing_id))
    const appendedListings = filteredListings.map((listing, i) =>         
        <ListingCard key={i} addSelectedListing={addSelectedListing} {...listing} {...listing.details} areaId={selectedAreaId}/>
    )
    return (
        <section className='listings-pg'>
            <div className='listings-header'>
                <h2 className='area-title'>Favorites</h2>
            </div>
            <h2 className='listing-count'>{filteredListings.length} favorites</h2>
            <section className='listings-container'>
                {favoriteListings.length === 0 ? <p className='no-favorites'>There are no favorited listings.</p>:appendedListings}
            </section>
        </section>
    )
}

export default Favorites;   

Favorites.propTypes = {
    addSelectedListing: PropTypes.func, 
    selectedAreaId: PropTypes.number, 
    listings: PropTypes.array, 
    favoriteListings: PropTypes.array
}
