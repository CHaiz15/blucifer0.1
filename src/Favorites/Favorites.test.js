import React from 'react';
import Favorites from './Favorites';
import { shallow } from 'enzyme'

describe('Favorites', () => {
  it('should be an instance of Favorites component', () => {
    const wrapper = shallow(
    <Favorites
      addSelectedListing={jest.fn()}
      listings={[]}
      favoriteListings={[]}
      selectedAreaId={1}
    />);
      expect(wrapper).toMatchSnapshot();
  });
})