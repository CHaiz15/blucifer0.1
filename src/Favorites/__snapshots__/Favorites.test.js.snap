// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites should be an instance of Favorites component 1`] = `
<section
  className="listings-pg"
>
  <div
    className="listings-header"
  >
    <h2
      className="area-title"
    >
      Favorites
    </h2>
  </div>
  <h2
    className="listing-count"
  >
    0
     favorites
  </h2>
  <section
    className="listings-container"
  >
    <p
      className="no-favorites"
    >
      There are no favorited listings.
    </p>
  </section>
</section>
`;
